### ISSUE TITLE
Feature request: Exceptions to no-if and no-conditional-expect for loops

### ISSUE THREAD

**Issue by Ariane-B** [jest-community/eslint-plugin-jest#1047]
We sometimes like to do looped tests, when we have a lot of similar things we want to test with only slightly different entry parameters. In these, it's okay to have conditions. The data for each test will be conditional, after all.

The simplest case is when we want to put a different word in the title depending on what the test requires, like this:

```javascript

      const answerPreviewTests = [
         {
            showInactive: true,
            isActive: true,
            expectedExistence: true,
         },
         // ...
      ]
      // It's okay if it's a loop
      /* eslint-disable jest/no-if */
      for (const testData of answerPreviewTests) {
         it(`SHOULD ${ testData.expectedExistence ? 'show' : 'hide' } the answer preview element WHEN showInactive is ${ testData.showInactive } and choice.isActive is ${ testData.isActive }`, async () => {
            // ...
         })
      }
```

But in all non-looped tests, conditions are a bad practice and should be flagged.

Would it be possible to add settings to those rules to allow exceptions for cases like these?

**Comment by G-Rath** [comment]
What you're doing is exactly what [`it.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) is for - is there any reason why you're not using that?

**Comment by Ariane-B** [comment]
Holy SHIT that's a thing?!

Then I'm changing my suggestion: Add a (recommended!) rule that tells people to use `it.each` whenever it sees a loop! For ignorant teams like us.

Dang, it's so cool that that thing exists.

**Comment by G-Rath** [comment]
Glad I could help - I've created #1048 to think about creating a new rule, and will close this :)
